_source/_posts/2020-12-18-how-to-use-webauthn-csharp-dotnet.md

```diff
@@ -0,0 +1,780 @@
+    try {
+            response = await registerCredentialWithServer(data);
+        } catch (e) 
+        {
+            alert(e);
+            return;
+        }
+
+  console.log("Credential Object", response);
+
+  // show error
+    if (response.status !== "ok") {
+        alert(response.errorMessage);
+        return;
+    }
+
+    alert("You've registered successfully.");
+}
+
+async function registerCredentialWithServer(formData) {
+
+    let response = await fetch('/Account/SaveCredentials', {
+        method: 'POST',
+        body: JSON.stringify(formData),
+        headers: {
+            'Accept': 'application/json',
+            'Content-Type': 'application/json'
+        }
+    });
+
+    let data = await response.json();
+
+    return data;
+}
+```
+
+Let's go through the code and break down what it does. First, I register an event handler for the `sign-up` form. In the event handler, I call the `/Account/CredentialOptions` endpoint with the user data and use the resulting options to call the `navigator.credentials.create` method. After that, I send the generated credentials to the server by calling the `/Account/SaveCredentials` endpoint. It's that simple.
+
+The `SaveCredentials` endpoint doesn't exist yet, but before I create that method, I will configure Okta as I will store user data and their credentials in Okta.
+
+## Store User and User's Public Key in Okta
+
+This tutorial will use Okta to store users and the public key of their credentials. Okta supports user authentication, multi-factor authentication, social authentication, as well as all OpenID Connect flows out of the box—it can take care of all the scenarios you might ever need.
+
+To get started, log in to your [Okta Developer account](https://developer.okta.com/) via the URL in the sign-up email to access your org's portal dashboard. Once you log in, make sure to note your Org URL from the upper right corner of the screen:
+
+{% img blog/webauthn-csharp/image6.png alt:"" width:"800" %}{: .center-image }
+
+Alternatively, you can install [Okta CLI](https://cli.okta.com/) and login to your Okta account with the `okta login` command or sign up for a new account with the `okta start` command if you don't already have an Okta account.
+
+Now, I will create a custom field for storing the user's public key. Go to **Users** in the menu, click **Profile Editor**, and click the **Profile** button next to the default Okta profile. Click **Add Attribute**, fill the `Add Attribute` form and click **Save**:
+
+{% img blog/webauthn-csharp/image8.png alt:"" width:"800" %}{: .center-image }
+
+Do the same to add another attribute called `CredentialId`.
+
+Next, go to **API** in the menu, click **Tokens**, and click the **Create Token** button. Enter the token name (I will use *OktaWebAuthn*) and click the **Create Token** button. Make sure you copy the token because you won't be able to view it again:
+
+{% img blog/webauthn-csharp/image9.png alt:"" width:"800" %}{: .center-image }
```

Comment:
```suggestion
{% img blog/webauthn-csharp/image9.png alt:"" width:"512" %}{: .center-image }
```
Add alt value. You also might want to make a new screenshot to replace this image. It's pretty grainy.